docs/RBAC.md

```diff
@@ -0,0 +1,152 @@
+
+The search service collects data using a service account with wide cluster access and stores all resources in the database. The Search API must enforce that results for each user (or service account) only contain resources that they are authorized to access.
+
+## [Swimlane diagram (link)](https://swimlanes.io/#tVZNb9tGEL3zVwyQiy2YlJ0URSE4BvwBpDokaGUnORgBvCaH0tbULrO7lOC0/e99u0uKpOwYOaSAQYvc+Xwzb2addBXP6JqFyVd0/secFhfnl0ny0bKh9GxwsBNa8NeGrUuStzOaTE4yOm/cipWTuXBMDRQnkyRR2sHugp2RvGGCBDn9wIpKo9fh1UQ7tGJRwJk2lGv9IDmjJBnEgyAuRQ75g0rnojqc0SdRycL72hnNOn+fmfIg3PvbRGmp4Ro+BNmVNo5qNlIXdPDrMVnOtSrsIRQQVNHkCOHG61oShlv5Sm64IKtpy7RuELbhTRfHWkNMlw6B0M1KWnJyzanTqVfqPOGzVAXXjIdyEA8xhmgLMk3F9oi2K4msIVlptQQmwtHJMa2lahxbryKVyJ3cSPeIlGU5o3nZpon8u1RhT6pgvRYItM8wI0A7ApNSwPvQ3HMqajl7WskO1x5y7+yIluzCORyV+oiEKghW8wcSsKCN/BYBB6CQgntfTUQ8odsA7II3kre+vF8OVs7Vdjad+iiMYiSaST0tdG6nhks2rHKeLlmx8ZkNxHzI081J9vrN9FUIywSr6eYkFX0eCCN9+M2mUh96976v54j5J/r2OHgYXnS8AzCiZndAoVgtSr5gdtf3WYJGeVKtMSGH5Sr6eozpAx2f9Iw+qoHPg1+OTw5bCr/O6J2AJbNXvdCUgTQdqV+k5WVogEi/Tqm1gcxF37B7B6F1m9rzJSCE3g1Gsr3kT0e92k6hgbHYqwOAjgadfTtfg4dWq64GP1h8kedsbairCs9v03GNY/1TuTOPb2GSLCOkeEgThmqMMrCghdvjaNjqxsAJ0Z3OCcGm/acU40YWTP/Q0nBNW+GQWftSSevuelu10X9x7qAT7PSvqbBvT32MZ1769pqr8rq594fnIbfIxZ/EBQvrNlqPyI05ueutjhmhUfIK8xQY2lzXfhZ26WMRjANeeAD/r3hDdX4kXCXWbGuRj0JNIl0HBDlN9wmyGPFpn9u+vZ9w6u9SWPRDzdgEpUir5b/9JAkkeannn1G+6VaOH0HYCQ6rRTe2ehzvDoE/+6jyldEKx76bfDGzJKW52oxGmYuLMgKHsYYtaIYr5xmVSP0o+f31ltLn0O49G9wKieS6qQoolyhZpJVfPrJ3EY+QR8wzjh6wYqEr7v5fYBNLtfSvl7H5utPB60DondFN7X986GofZBdX7QR9k9GfDZvHMMO9x3th2Q/MVzTqiKdDc4HLT5sB3NYGe7Udu/6a9J0SX7UeZsl8p+U9D9GFya9PQspAqPPL9/R7c9/jOsPkuQuzBy0PlyqVYbjQaSdyNpwidJsqOlX27Au0YO+9UGLp4Y7IvWw3N+xr1Oq0YH/ydwFvtDXhQR65hBnqbiKTyc3F1WwyoQ8cV6coingBK9gJWSH1e93gVqm3/qq2lVX1DBCAG858ecPn0EelrOA82yPxWY/2MzXeX4lx0y7QI2h+OHehf5vKDXmf/Ac=)
+
+![](https://static.swimlanes.io/2ee006a19f5dee69ece614e95f72d480.png)
+
+
+## Protect the Search API
+<!-- [NEW IN V2] This feature is new for V2. -->
+The API itself is protected by RBAC. Users must be given a role that allows access to search.
+
+The default ACM admin and viewer roles should include access to the Search API by default.
+
+> **OPEN DISCUSSION:**
+>
+> We are still investigating the implementation of this feature. We'll update as we learn more.
+>
+> Options:
+> 1. Kube API server extension - This can add additional load to the kube API server.
+> 2. Management Ingress - This is being deprecated.
+> 3. Validate access at the service - Provides the most flexibility, but how will it leverage kubernetes implementation?
+
+## Authenticate the user 
+
+The Search API authenticates the user (or service account) and impersonates the user to obtain their access rules.
+We cache the token validation for a short period (60 seconds) to reduce requests to the kube api. Tokens have a short expiration, so we must revalidate often. **Note that** this time-to-live period is independent of the cached rules, which is longer at 10 minutes of inactivity.
+
```

Comment:
Is this validation time 60s - this is configurable ?

Reply:
Updated to `(default: 60 seconds)` to be a little clearer.